services/storage/predicate.go

```diff
@@ -0,0 +1,152 @@
+package storage
+
+import (
+	"bytes"
+	"strconv"
+)
+
+// NodeVisitor can be called by Walk to traverse the Node hierarchy.
+// The Visit() function is called once per node.
+type NodeVisitor interface {
+	Visit(*Node) NodeVisitor
+}
+
+func (x Node_Type) MarshalJSON() ([]byte, error) {
```

Comment:
Reason for the underscore syntax?

Reply:
The underscore comes from the protobuf compiler. I'm not using the `MarshalJSON`, so I'll remove it.